src/libraries/Lockers.sol

```diff
@@ -8,6 +8,9 @@ library Lockers {
     // The starting slot for an array of lockers, stored transiently.
     uint256 constant LOCKERS_SLOT = uint256(keccak256("Lockers")) - 1;
```

Comment:
e.g. i'd like to be sure that:
- you start with a nested lock, length 2
- the second locker gets a `currentHook` of X
- second locker releases (and is popped, length = 1)
- a new second locker takes out a lock (length = 2) and in that moment `currentHook == address(0)` and there isnt some remnant currentHook from before the decrement of `length`

Reply:
Interesting.. so it is only true OUTSIDE of the library. (ie because we manually clear it any time a hookAddr is pushed, not because the library enforces this). 

This means we can only test that this holds true outside of the library unit tests

Reply:
added this test here! https://github.com/Uniswap/v4-core/pull/404/commits/f70443aabe3308cadcd6130b79b0a4da827fef06